packages/blade/src/components/Button/ButtonSpinner/ButtonSpinner.web.tsx

```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import styled from 'styled-components';
+import useIconProps from '../../Icons/useIconProps';
+import BaseSpinner from '../../Spinner/BaseSpinner';
+import type { ButtonSpinnerProps } from './ButtonSpinner.d';
+import { buttonSpinnerStyles, buttonSpinnerContainerStyles } from './buttonSpinnerStyles';
+
+const ButtonBaseSpinner = styled(BaseSpinner)(buttonSpinnerStyles);
+
+const ButtonSpinnerContainer = styled('div')<{ isHidden: boolean }>(({ isHidden }) => ({
+  opacity: isHidden ? 0 : 1,
+  ...buttonSpinnerContainerStyles,
+}));
+
+const ButtonSpinner = ({
+  isLoading,
+  children,
+  color,
+  size,
+}: ButtonSpinnerProps): React.ReactElement => {
+  const { width, iconColor } = useIconProps({ size, color });
+
+  return (
+    <>
+      {isLoading ? <ButtonBaseSpinner color={iconColor} size={width} /> : null}
```

Comment:
> why can't we conditionally render the entire thing? the container and spinner?

We can't conditionally render it. We need the element to be present on DOM but only hidden with CSS otherwise the width won't stay same.

Reply:
> Also do we explicitly need container? can't we just use Button's container? and just add spinner to that?

can you elaborate the JSX? Could not understand it.